### PR TITLE
Ch 8 exercises

### PR DESCRIPTION
**Pull request by Tillaert**
Partial exercises for chapter 8.

### CODE REVIEW

File: src/test/scala/nl/hugo/redbook/Spec.scala [hunk 1]
```diff
@@ -2,6 +2,6 @@ package nl.hugo.redbook
 
 import org.scalatest.{ Matchers, WordSpec }
 
-trait Spec extends WordSpec with Matchers {
+class Spec extends WordSpec with Matchers {
```
Comment (hzwaal) [inline src/test/scala/nl/hugo/redbook/Spec.scala]: Why this change?

Comment (Tillaert) [inline src/test/scala/nl/hugo/redbook/Spec.scala]: It wouldn't compile without this change. I'll investigate it a bit more when I have the time.
